src/components/spreadsheet/utils/equipement-table-popup-editors.js

```diff
@@ -0,0 +1,295 @@
+import React, {
+    useCallback,
+    useState,
+    forwardRef,
+    useImperativeHandle,
+    useMemo,
+} from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import DeleteIcon from '@mui/icons-material/Delete';
+import AddCircleIcon from '@mui/icons-material/AddCircle';
+import {
+    Box,
+    Grid,
+    IconButton,
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TableRow,
+    TextField,
+    Tooltip,
+    useTheme,
+} from '@mui/material';
+import { SelectOptionsDialog } from 'utils/dialogs';
+import yup from 'components/utils/yup-config';
+
+const validationSchema = yup
+    .array()
+    .of(
+        yup.object().shape({
+            key: yup.string().required('FillAllFields'),
+            value: yup.string().required('FillAllFields'),
+        })
+    )
+    .test('unique-keys', 'DuplicatedProps', (values) => {
+        const keys = values.map((value) => value.key);
+        const uniqueKeys = new Set(keys);
+        return keys.length === uniqueKeys.size;
+    });
+
+function arrayToObject(arr) {
+    return arr.reduce((obj, item) => {
+        obj[item.key] = item.value;
+        return obj;
+    }, {});
+}
+
+/**
+ * @author Jamal KHEYYAD <jamal.kheyyad at rte-international.com>
+ */
+export const SitePropertiesEditor = forwardRef(
+    ({ gridContext, colDef, gridApi, rowData }, ref) => {
+        const theme = useTheme();
+        const [error, setError] = useState('');
+        const intl = useIntl();
+        const [open, setOpen] = useState(true);
+        const [localRowData, setRowData] = useState(() => {
+            const data = rowData;
+            if (!data?.properties) {
+                return [];
+            }
+            const keys = Object.keys(data.properties);
+            return keys.map((key, index) => {
+                return { id: index, key: key, value: data.properties[key] };
```

Comment:
May be it's better to extract this to a single method then use it here to initiate the state.